WordPress/Classes/ViewRelated/Post/PostCell.swift

```diff
@@ -0,0 +1,271 @@
+import UIKit
+import Gridicons
+
+class PostCell: UITableViewCell, ConfigurablePostView {
+    @IBOutlet weak var featuredImageStackView: UIStackView!
+    @IBOutlet weak var featuredImage: CachedAnimatedImageView!
+    @IBOutlet weak var featuredImageHeight: NSLayoutConstraint!
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var snippetLabel: UILabel!
+    @IBOutlet weak var dateLabel: UILabel!
+    @IBOutlet weak var authorLabel: UILabel!
+    @IBOutlet weak var statusLabel: UILabel!
+    @IBOutlet weak var statusView: UIStackView!
+    @IBOutlet weak var progressView: UIProgressView!
+    @IBOutlet weak var editButton: UIButton!
+    @IBOutlet weak var retryButton: UIButton!
+    @IBOutlet weak var viewButton: UIButton!
+    @IBOutlet weak var moreButton: UIButton!
+    @IBOutlet weak var actionBarView: UIStackView!
+    @IBOutlet weak var containerView: UIView!
+    @IBOutlet weak var upperBorder: UIView!
+    @IBOutlet weak var bottomBorder: UIView!
+    @IBOutlet weak var topPadding: NSLayoutConstraint!
+    @IBOutlet weak var contentStackView: UIStackView!
+    @IBOutlet weak var titleAndSnippetView: UIStackView!
+    @IBOutlet weak var topMargin: NSLayoutConstraint!
+
+    private let margin: CGFloat = WPDeviceIdentification.isiPad() ? 20 : 16
+    private let titleTopMargin: CGFloat = WPDeviceIdentification.isiPad() ? 6 : 2
+    private let featuredImageHeightConstant: CGFloat = WPDeviceIdentification.isiPad() ? 226 : 100
+
+    lazy var imageLoader: ImageLoader = {
+        return ImageLoader(imageView: featuredImage, gifStrategy: .mediumGIFs)
+    }()
+
+    private var post: Post!
```

Comment:
Can we do this without the `!`s?

Reply:
By making it `Optional` we will need to add boilerplate `guard` in 9 different places. Also, all the methods that use `post` are called only after a post is assigned.

What do you think? Keep the `!` or change it to `?`?

Reply:
I'm solidly in the `?` camp. This is something our team has debates both sides for—sadly you're caught on my side at the moment 😆

Reply:
Done!